src/Security/Voter/SignalementVoter.php

```diff
@@ -137,34 +149,70 @@ private function canDelete(Signalement $signalement, User $user): bool
         return $this->isAdminOrTerritoryAdmin($signalement, $user);
     }
 
-    private function canEdit(Signalement $signalement, User $user): bool
+    private function canEditStatus(Signalement $signalement, User $user, SignalementStatus $status, ?bool $checkAffectationAccepted = true): bool
     {
         if ($this->security->isGranted('ROLE_ADMIN')) {
             return true;
         }
-        if (SignalementStatus::CLOSED === $signalement->getStatut() && $user->isTerritoryAdmin() && $user->hasPartnerInTerritory($signalement->getTerritory())) {
-            return true;
-        }
-        if (!in_array($signalement->getStatut(), [SignalementStatus::ACTIVE, SignalementStatus::INJONCTION_BAILLEUR])) {
+        if ($status !== $signalement->getStatut()) {
             return false;
         }
         if ($user->isTerritoryAdmin() && $user->hasPartnerInTerritory($signalement->getTerritory())) {
             return true;
         }
+        if (!$checkAffectationAccepted) {
+            return false;
+        }
         $partner = $user->getPartnerInTerritory($signalement->getTerritory());
 
         return $signalement->getAffectations()->filter(function (Affectation $affectation) use ($partner) {
             return $affectation->getPartner()->getId() === $partner?->getId() && AffectationStatus::ACCEPTED === $affectation->getStatut();
         })->count() > 0;
     }
 
+    private function canEdit(Signalement $signalement, User $user): bool
+    {
+        return $this->canEditStatus($signalement, $user, SignalementStatus::ACTIVE);
+    }
+
+    private function canEditInjonction(Signalement $signalement, User $user): bool
+    {
+        if ($this->canEditStatus($signalement, $user, SignalementStatus::ACTIVE)) {
+            return true;
+        }
+
+        return $this->canEditStatus($signalement, $user, SignalementStatus::INJONCTION_BAILLEUR);
```

Comment:
On en a parlé sur Mattermost, je le remet là, je préfèrerais mettre `if is_granted('SIGN_EDIT') or is_granted('SIGN_EDIT_INJONCTION')` côté vérification et que le voter soit juste `return $this->canEditStatus($signalement, $user, SignalementStatus::INJONCTION_BAILLEUR);`

Ce serait un peu plus long, mais plus lisible côté vérification, et moins contre-intuitif côté Voter